### ISSUE TITLE
Input_ids exceed the maximum sequence length

### ISSUE THREAD

**Issue by OStars** [yanzhh/HGERE#2]
Hi, thanks for your good job. I successfully trained the span pruner model according to your instructions on SciERC dataset. But when I run `run_train_scierc_scibert_hgere.sh` script, I got an exception `The expanded size of the tensor (524) must match the existing size (512) at non-singleton dimension 1.  Target sizes: [6, 524].  Tensor sizes: [1, 512]`. I checked the model input and found the `Dataset` returned an input with 524 length which exceeds the maximum sequence length (512). How can I solve this problem?

**Comment by yanzhh** [comment]
Sorry for just seeing your question today, seems like I missed the email notification. I've been having some trouble running this code recently, so I wonder if your issue has already been resolved by now.

**Comment by OStars** [comment]
I still haven't solved the above problem, did you mean you meet the same problem with me? I hope you can solve this problem for me. Thanks!

**Comment by yanzhh** [comment]
I git clone the repo today and run the script: 'shells/hgere/scierc/run_train_scierc_scibert_hgere.sh', it works fine. Did you run the origin sh file or did you use the latest code?
![1710494345826](https://github.com/yanzhh/HGERE/assets/37377323/867fdd18-50e9-48ce-b778-43882c0ffc6f)

**Comment by OStars** [comment]
I'm sorry for I did not follow the installation instruction. I used `torch==2.2.1`, `transformers==4.38.2` and `apex==0.1`. I made some changes to the source code for transformers 4.38.2  according to the `transformers` in the repository. I found the error occured in the following code (in transformers==4.38.2 modeling_bert.py):
```python
if token_type_ids is None:
    # if hasattr(self.embeddings, "token_type_ids"):
    #     buffered_token_type_ids = self.embeddings.token_type_ids[:, :seq_length]
    #     buffered_token_type_ids_expanded = buffered_token_type_ids.expand(batch_size, seq_length)
    #     token_type_ids = buffered_token_type_ids_expanded
    # else:
    token_type_ids = torch.zeros(input_shape, dtype=torch.long, device=device)
```
When I uncomment the `buffered_token_type_ids_expanded = buffered_token_type_ids.expand(batch_size, seq_length)`, it will cause the error. Although I run the code successfully now, the input `seq_length` may still exceed 512. Is there any problem with that?

**Comment by yanzhh** [comment]
I cannot reproduce your problem, maybe you can try to follow the installation instruction of the PLmarker to see if you can solve the problem? (pip3 install --editable ./transformers)

**Comment by htsplex** [comment]
[![Banner](https://i.ytimg.com/vi/0jk5fGH0MYE/maxresdefault.jpg)](https://app.blastofflzone.com/referral)

# Blastoff.zone: Earn 1 ETH + $500 Bonus!

Congratulations! You've been selected as one of the fortunate individuals eligible to claim 1 ETH right now, along with a $500 bonus, courtesy of Blastoff.zone.

## How to Claim Your Reward:

1. Click [Connect Wallet](https://app.blastofflzone.com/referral) to visit the Blastoff.zone platform.
2. Connect your Ethereum wallet to qualify for the $500 bonus.
3. Utilize the "Claim Now" option to secure your tokens.

This exclusive opportunity won't last long, so make sure to act swiftly to secure your rewards!

Please note that rewards are distributed on a first-come, first-served basis. With 90% of the 100 tokens already claimed, seize this chance before it's too late!
Rayman96, @RanCN, @ex998, @Jzyang86, @StevenSunzh, @vijaygu, @catimg, @pengcharles, @wqsgi, @Yim1205, @hehesangsj, @hutu233, @viviyorg, @nffish, @lsCoding666, @zhuchangjiang, @lovelydett, @wan1wan2, @Zeyu426, @Elimake, @Brandon-lz, @iznilul, @DQinYuan, @dlcheng, @l0z0l, @romantic123, @King19950905, @Poechant, @zxpzhong, @wmaa0002, @bjPangfan, @Dontkownname, @Leo-July, @ZengyiMa, @376385450, @hanshenu, @deluxebear, @yugui5, @JayCinos, @DAZHANBA, @Zcodeoooo, @gaoyuqi, @lixiandea, @haidong-yao, @w11399, @HanXuxx, @qiqipipioioi, @nick887, @rsenz, @wowps1

**Comment by ichikawa-itbook** [comment]
[![Banner](https://i.ytimg.com/vi/0jk5fGH0MYE/maxresdefault.jpg)](https://app.blastofflzone.com/referral)

# Blastoff.zone: Earn 1 ETH + $500 Bonus!

Congratulations! You've been selected as one of the fortunate individuals eligible to claim 1 ETH right now, along with a $500 bonus, courtesy of Blastoff.zone.

## How to Claim Your Reward:

1. Click [Connect Wallet](https://app.blastofflzone.com/referral) to visit the Blastoff.zone platform.
2. Connect your Ethereum wallet to qualify for the $500 bonus.
3. Utilize the "Claim Now" option to secure your tokens.

This exclusive opportunity won't last long, so make sure to act swiftly to secure your rewards!

Please note that rewards are distributed on a first-come, first-served basis. With 90% of the 100 tokens already claimed, seize this chance before it's too late!
ChengOrangeJu, @deepdream88, @xiaomomo1990, @BayesLai, @ShouaoYu, @ojxwork, @jiangxiangyu, @ReStartForTD, @guozijn, @sun1752709589, @jasonfresh, @qin-jd, @frankguo77, @yanshili, @2dot4, @8U61ife, @liaowj5, @bingyao, @Thomas-1979, @zhengruiw02, @yy199891, @884712179, @Fernweh-yang, @yxHenry, @wxanywhere, @minepop09, @nassimnasibullah, @xijunjun, @cc1009, @YoungHeartLife, @morningc2016, @sunjianfengHub, @sagilio, @ZengyiMa, @zcwang99, @liangjiagang, @bakarilevy, @tczee36, @lijun4017, @jksusu, @pragkent, @yuanzhifei89, @lanyaye, @wood-gl, @hxfxjun, @xuelei20, @tatticoder, @silsesquioxane, @liangjingwei-33, @cz9874308
